src/routes/email/exception-filters/email-does-not-exist.exception-filter.ts

```diff
@@ -0,0 +1,23 @@
+import {
+  ArgumentsHost,
+  Catch,
+  ExceptionFilter,
+  HttpStatus,
+} from '@nestjs/common';
+import { Response } from 'express';
+import { EmailAddressDoesNotExistError } from '@/datasources/email/errors/email-address-does-not-exist.error';
+
+@Catch(EmailAddressDoesNotExistError)
+export class EmailAddressDoesNotExistExceptionFilter
+  implements ExceptionFilter
+{
+  catch(exception: EmailAddressDoesNotExistError, host: ArgumentsHost) {
+    const ctx = host.switchToHttp();
+    const response = ctx.getResponse<Response>();
+
+    response.status(HttpStatus.NOT_FOUND).json({
+      message: 'The provided email address does not exist.',
```

Comment:
nitpick: probably a more generic message would be better since some routes do not expect the client to provide the email address.

Reply:
Changed in 27eaae4e907918ecba4248fe5b52aceb7c32a10e.